### PR TITLE
v1.7 backports 2020-12-03

### PR DESCRIPTION
**Pull request by joestringer**
* #14238 -- vendor: update etcd to 3.4.14 (@aanm)
  * Complete re-vendor, @aanm please look closely.

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 14238; do contrib/backporting/set-labels.py $pr done 1.7; done
```

### CODE REVIEW

**Comment by joestringer** [comment]
test-backport-1.7

File: go.mod [hunk 1]
```diff
@@ -71,14 +71,14 @@ require (
...
-	go.etcd.io/etcd v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738
+	go.etcd.io/etcd v0.5.0-alpha.5.0.20201125193152-8a03d2e9614b
 	golang.org/x/crypto v0.0.0-20200220183623-bac4c82f6975
 	golang.org/x/net v0.0.0-20191004110552-13f9640d40b9
 	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
 	golang.org/x/sys v0.0.0-20200124204421-9fbb57f87de9
 	golang.org/x/time v0.0.0-20191024005414-555d28b269f0
 	google.golang.org/genproto v0.0.0-20190502173448-54afdca5d873
-	google.golang.org/grpc v1.23.1
+	google.golang.org/grpc v1.26.0
```
Comment (joestringer) [inline go.mod:81]: This one in particular seems a bit fishy. I don't actually see any grpc changes here but running the scripts caused this line to change.

Comment (aanm) [reply go.mod:81]: this means that etcd might depend on grpc `1.26` and the compilation will fail because the `replace` was not modified.

Comment (aanm) [reply go.mod:81]: I think it's wise to drop this PR as users haven't complain about it...

**Comment by joestringer** [comment]
Looks like this needs a bit more attention to properly merge. Unless we specifically hear something about this issue against v1.7, it seems prudent to just leave it out. This will be fixed in the next 1.8.x release, so there will also be an option for affected users to address the underlying issue.

**Pull request closed without merging**